js/modules/loki_file_server_api.js

```diff
@@ -40,6 +43,184 @@ class LokiFileServerAPI {
     return this._setOurDeviceMapping(authorisations, isPrimary);
   }
 
+  async getDeviceMappingForUsers(pubKeys) {
+    const users = await this._server.getUsersAnnotations(pubKeys);
+    return users;
+  }
+
+  async verifyUserObjectDeviceMap(
```

Comment:
I am not sure to understand fully how this function works, but it looks like it could benefit from some simplification.
There's probably a way to simply return a map pubkey => (username-authorisation pair), or at least not having to pass the two predicates/callbacks `iterator` and `notFoundHandler`.
Then the caller can be responsible to iterate through the map and handle the "not found" case and the "iterator" case itself.

Reply:
happy to discuss it. I had two versions of the function before which was more readable but I felt they did such similar things that DRY principle was the higher value, so we don't fix a bug in one and not the other. This function only holds the common code used by the two callers, where it's caller set ups the unique changes each needs.

Reply:
iterator is scoped per user authorisation and notFoundHandler is per user